module.combodo-approval-extended.php

```diff
@@ -73,7 +78,28 @@ class ApprovalExtendedInstaller extends ModuleInstallerAPI
 	{
 		public static function BeforeWritingConfig(Config $oConfiguration)
 		{
-			// If you want to override/force some configuration values, do it here
+			if (empty(utils::GetConfig()->GetModuleSetting('combodo-approval-extended', 'targets', null))
+					&& !empty(utils::GetConfig()->GetModuleSetting('combodo-approval-extended', 'target_state', null) )) {
+				// Migration of the old configuration to the new one
+				$newConfiguration = [
+													'UserRequest' =>   [
+														'target_states' => [$oConfiguration->GetModuleSetting('combodo-approval-extended', 'target_state', null) ],
+														'bypass_profiles' =>$oConfiguration->GetModuleSetting('combodo-approval-extended', 'bypass_profiles', 'Service Manager') ,
+														'reuse_previous_answers' => $oConfiguration->GetModuleSetting('combodo-approval-extended', 'reuse_previous_answers', true) ,
+													]
+												];
+				$oConfiguration->SetModuleSetting('combodo-approval-extended', 'targets', $newConfiguration);
+			}
+			// Replacing old conf parameters value to indicate that it is obsolete
+			$aParamsToRemove = array('target_state', 'bypass_profiles', 'reuse_previous_answers');
+			foreach($aParamsToRemove as $sParamToRemove)
+			{
+				$sParamCurrentValue = $oConfiguration->GetModuleSetting('combodo-approval-extended', $sParamToRemove, null);
+				if(!empty($sParamCurrentValue))
+				{
+					$oConfiguration->SetModuleSetting('combodo-approval-extended', $sParamToRemove, 'No longer used, you can remove this parameter.');
```

Comment:
Why not just set them to `null`? Or remove the setting?

Reply:
This is to keep previous parameters for tests and check migration

Reply:
Yeah, so it is indeed best to just leave the parameter and not set it to a specific string..